src/lib.rs

```diff
@@ -179,36 +189,55 @@ use std::{
 /// Execute child processes. See the module documentation on how to use it.
 #[macro_export]
 macro_rules! cmd {
-    ($($args:expr),+) => {{
+    ($($args:tt)*) => {{
         let context = $crate::Context::production();
-        $crate::panic_on_error($crate::cmd_result_with_context!(context, $($args),+))
+        $crate::panic_on_error($crate::cmd_result_with_context!(context, $($args)*))
     }}
 }
 
 /// Like [`cmd!`], but fixes the return type to `()`.
 #[macro_export]
 macro_rules! cmd_unit {
-    ($($args:expr),+) => {{
-        let () = $crate::cmd!($($args),+);
+    ($($args:tt)*) => {{
+        let () = $crate::cmd!($($args)*);
     }}
 }
 
 /// Like [`cmd!`], but fixes the return type to [`Result<T, Error>`],
 /// where `T` is any type that implements [`CmdOutput`].
 #[macro_export]
 macro_rules! cmd_result {
-    ($($args:expr),+) => {{
+    ($($args:tt)*) => {{
         let context = $crate::Context::production();
-        $crate::cmd_result_with_context!(context, $($args),+)
+        $crate::cmd_result_with_context!(context, $($args)*)
     }}
 }
 
+#[doc(hidden)]
+#[macro_export]
+macro_rules! prepare_config {
+    ($config:ident # @ $head:expr, $($tail:tt)*) => {
```

Comment:
One thing I thought of to make this clearer would be to use `config: …, args: …` instead of `#` to document what's going on. Like so:

```rust
    (config: $config:ident, args: @ $head:expr, $($tail:tt)*) => {
```

This would make it a little clearer what's happening, although would be a little longer.

Reply:
Yeah, I really like that! I was trying to make the macro invocation syntax a bit clearer by introducing the `#`, but wasn't really happy with that. Didn't occur to me to use named arguments.